### DISCUSSION TITLE
Fetch failed during deployment to vercel

### DISCUSSION THREAD

**Discussion by Sabari2810** [vercel/next.js#58018] [Help]
### Summary

![image](https://github.com/vercel/next.js/assets/76817070/53ee38bf-b505-467e-8248-8ccd2ec4f342)

I have a NextJS application and I am using Next.js API routes as API, one of the pages in my project is a server-side component and it calls the API to get the page information. It works on my local machine since the API would be running, but it fails during deployment, because the fetch is failing.

**Comment by paulrogov** [comment]
Here is a response by Next developer:
https://github.com/vercel/next.js/issues/44062#issuecomment-1653822615

In short:  
add this line at the beginning of a page that fails to be built  
`export const dynamic = 'force-dynamic'`

It works at least for now (Next 14, Node v20).

**Reply by zineanteoh** [reply]
what is the reason adding `force-dynamic` fixes this?

**Reply by paulrogov** [reply]
By adding `force-dynamic` you explicitly tell the Next builder that this page is _dynamic_ and Next must not try to generate a static version of it.
 
Otherwise the build breaks with a `fetch error` as mentioned above in the screenshot, because the content of this page is not available at the build time (the page is a server-side component)
